client/mainwindow.cpp

```diff
@@ -9,37 +10,38 @@
 
 MainWindow::MainWindow(QWidget* parent)
   : QMainWindow(parent)
-  , ui(std::make_unique<Ui::MainWindow>())
 {
-    ui->setupUi(this);
-    ui->lineEdit->setPlaceholderText("Enter what you want send to server here...");
-    ui->sendButton->setDisabled(true);
-    ui->disconnectButton->setDisabled(true);
-    ui->lineEdit->setDisabled(true);
-    textBrowser = QPointer<QTextBrowser>(ui->textBrowser);
-    connection = std::make_shared<Connection>(textBrowser);
-    connect(ui->connectButton, &QPushButton::clicked, this, &MainWindow::onConnectClicked);
-    connect(ui->disconnectButton, &QPushButton::clicked, this, &MainWindow::onDisconnectClicked);
-    connect(ui->sendButton, &QPushButton::clicked, this, &MainWindow::onSendClicked);
+    main_tab_widget = new QTabWidget();
+    connection_tab = std::make_unique<Connection_tab>(this);
+    processes_tab = std::make_unique<Processes_tab>(this);
+
+    main_tab_widget->addTab(connection_tab->get_tab_widget(), "Connection");
+    main_tab_widget->addTab(processes_tab->get_tab_widget(), "Processes");
```

Comment:
`connection_tab` and `process_tab` are widgets. They could be embedded as tabs inside `main_tab_widget`. There is no need to create separate child widget to embed it.